snuba/utils/streams/kafka.py

```diff
@@ -83,6 +88,16 @@ def value(self) -> bytes:
     def headers(self) -> Headers:
         return self.__headers
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, KafkaPayload):
+            return False
+        else:
+            return (
+                self.key == other.key
+                and self.value == other.value
+                and self.headers == other.headers
+            )
```

Comment:
This isn't used in this change yet, but it will be as soon as `KafkaStreamsTestCase.get_payloads` returns an `Iterator[KafkaPayload]` instead of `Iterator[int]`.